### PR TITLE
DAT-71446: allow escaped quotes in string values

### PR DESCRIPTION
**Pull request by makc-dataloop**
Added a bunch of tests and made sure they pass, as well as manual input or selecting from suggestions popup.

### CODE REVIEW

File: src/hooks/use-suggestions.ts [hunk 1]
```diff
@@ -614,10 +616,19 @@
...
+    }
+    const matchDoubleQuotes = str.match(/^\s*(?<!\\)"(.*)(?<!\\)"\s*$/)
+    if (matchDoubleQuotes) {
+        return !/(?<!\\)"/.test(matchDoubleQuotes[1])
+    }
+    return false
+}
+
+const enquoteString = (str: string) => {
+    return "'" + str.replace(/'/g, "\\'") + "'"
```
[the remark this answers is not in the document]
Comment (makc-dataloop) [reply src/hooks/use-suggestions.ts]: closing as per discussion with @guyDataloop

File: src/utils/parse-smart-query.ts [hunk 2]
```diff
@@ -112,66 +131,67 @@ export const parseSmartQuery = (
...
-                        ;[key, value] = term
-                            .split('NOT-IN')
-                            .map((x) => x.trim())
-                        queryValue = term
-                            .split('NOT-IN')
-                            .map((x) => x.trim())[1]
+                        ;[key, value] = SplitFirst(term, 'NOT-IN').map((x) =>
+                            x.trim()
+                        )
+                        // TODO make , inside quotes work
```
Comment (guyDataloop) [inline src/utils/parse-smart-query.ts:193]: What is this TODO for ? Please create a ticket for it if needed

File: src/utils/parse-smart-query.ts [hunk 3]
```diff
@@ -181,9 +201,8 @@ export const parseSmartQuery = (
                             andQuery.push({ [key]: { $nin: pureValue } })
                         }
                     } else {
-                        queryValue = term
-                            .split('IN')
-                            .map((x) => x.trim())[1]
+                        // TODO make , inside quotes work
```
Comment (guyDataloop) [inline src/utils/parse-smart-query.ts:204]: What is this TODO for ? Please create a ticket for it if needed

**Review by guyDataloop — CHANGES_REQUESTED** [review]

**Review by fadiDL — APPROVED** [review]

File: src/utils/parse-smart-query.ts (hunk 2 above)
Comment (makc-dataloop) [reply src/utils/parse-smart-query.ts:193]: sure, https://github.com/dataloop-ai/components/issues/943

File: src/utils/parse-smart-query.ts (hunk 3 above)
Comment (makc-dataloop) [reply src/utils/parse-smart-query.ts:204]: just an unrelated issue I have noticed while working on this section of the code (created https://github.com/dataloop-ai/components/issues/943)

**Review by guyDataloop — APPROVED** [review]

**Pull request merged**